tests/test_ast.rs

```diff
@@ -234,3 +234,105 @@ fn test_hello_java() {
 ",
     );
 }
+
+#[test]
+fn test_non_existant_input() {
+    let lex = Path::new("grammars/calc.l");
+    let yacc = Path::new("grammars/calc.y");
+    let ast_result = parse_file("nosuchfileexists.calc", lex, yacc);
+    assert!(ast_result.is_err());
+    match ast_result {
+        Err(ParseError::FileNotFound(s)) => {
+            if s == String::from("nosuchfileexists.calc") {
```

Comment:
Why not just replace this whole `if` with `assert_eq!(s, "nosuchfileexists.calc")` or similar?